week2/homework/src/todo.js

```diff
@@ -0,0 +1,137 @@
+const fs = require('fs');
+
+const help = () => {
+  try {
+    console.log('Command - descriptions');
+    console.log('---------------------');
+    console.log(
+      '1. list - Shows current to-dos, or shows an appropriate text if there are no to-dos'
+    );
+    console.log('---------------------');
+    console.log(
+      '2. add - Adds a to-do item. All the words behind add are entered as 1 to-do item to the list.'
+    );
+    console.log('---------------------');
+    console.log('3. remove - Removes a to-do item by its 1-base index, e.g. to remove second item');
+    console.log('---------------------');
+    console.log('4. reset - Removes all to-do items from the list');
+    console.log('---------------------');
+    console.log('5. update - Updates a to-do item with new text');
+  }
+ catch (error) {
+    return console.log(error.message);
+  }
+};
+
+const fetchData = () => {
+  try {
+    const dataString = fs.readFileSync('todoData.json');
+    return JSON.parse(dataString);
+  }
+ catch (error) {
+    return [];
+  }
+};
+
+let todoArray = fetchData();
+const addTodo = title => {
+  try {
+    const todo = { title: title };
+    const isDuplicate = todoArray.filter(todo => todo.title === title);
+    if (isDuplicate.length === 0) {
+      todoArray.push(todo);
+      saveData(todoArray);
+      console.log('Todo item added');
+      console.log('---------------');
+      return todo;
+    }
+ else {
+      console.log('Todo already exists');
+    }
+  }
+ catch (error) {
+    return console.log(error.message);
+  }
+};
+
+const deleteTodo = index => {
+  try {
+    if (index > 0) {
+      const isFiltered = todoArray.filter(todo => todo.title !== todoArray[index].title);
+      saveData(isFiltered);
+      console.log('Deleted successfully!');
+      return todoArray.length !== isFiltered.length;
+    }
+ else {
+      console.log('Please enter available index!');
+    }
+  }
+ catch (error) {
+    return console.log(error.message);
+  }
+};
+
+const updateTodo = (index, todo) => {
+  try {
+    if (index >= 0) {
+      todoArray[index].title = todo;
+      saveData(todoArray);
+      console.log('Updated!');
+    }
+ else {
+      console.log('Please enter available index');
+    }
+  }
+ catch (error) {
+    return console.log(error.message);
+  }
+};
+
+const saveData = data => {
+  try {
+    fs.writeFileSync('todoData.json', JSON.stringify(data));
+  }
+ catch (error) {
+    return console.log(error.message);
```

Comment:
not that it is wrong but is it really necessary to catch the error here? Would it not be caught in updateTodo for example

Reply:
Yes you are right. No need to use here,..